 lib/curl_threads.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/curl_threads.h b/lib/curl_threads.h
index 0e067c0..daf3b87 100644
--- a/lib/curl_threads.h
+++ b/lib/curl_threads.h
@@ -41,7 +41,7 @@
 #  define curl_mutex_t           CRITICAL_SECTION
 #  define curl_thread_t          HANDLE
 #  define curl_thread_t_null     (HANDLE)0
-#  define Curl_mutex_init(m)     InitializeCriticalSectionEx(m, 0, 1)
+#  define Curl_mutex_init(m)     InitializeCriticalSection(m)
 #  define Curl_mutex_acquire(m)  EnterCriticalSection(m)
 #  define Curl_mutex_release(m)  LeaveCriticalSection(m)
 #  define Curl_mutex_destroy(m)  DeleteCriticalSection(m)
