 lib/getenv.c | 38 ++++++++++++++++++++++++++++----------
 1 file changed, 28 insertions(+), 10 deletions(-)

diff --git a/lib/getenv.c b/lib/getenv.c
index a957e6e..8502fdb 100644
--- a/lib/getenv.c
+++ b/lib/getenv.c
@@ -22,24 +22,21 @@
  *
  ***************************************************************************/
 #include "curl_setup.h"
+#include "curlx/curlx.h"
 
-char *curl_getenv(const char *variable)
+#if defined(_WIN32)
+static wchar_t *GetEnv(const wchar_t *variable)
 {
-#if defined(CURL_WINDOWS_UWP) || \
-  defined(__ORBIS__) || defined(__PROSPERO__) /* PlayStation 4 and 5 */
-  (void)variable;
-  return NULL;
-#elif defined(_WIN32)
   /* This uses Windows API instead of C runtime getenv() to get the environment
      variable since some changes are not always visible to the latter. #4774 */
-  char *buf = NULL;
-  char *tmp;
+  wchar_t *buf = NULL;
+  wchar_t *tmp;
   DWORD bufsize;
   DWORD rc = 1;
   const DWORD max = 32768; /* max env var size from MSCRT source */
 
   for(;;) {
-    tmp = curlx_realloc(buf, rc);
+    tmp = (wchar_t*)curlx_realloc(buf, rc * sizeof(wchar_t));
     if(!tmp) {
       curlx_free(buf);
       return NULL;
@@ -50,7 +47,7 @@ char *curl_getenv(const char *variable)
 
     /* it is possible for rc to be 0 if the variable was found but empty.
        Since getenv does not make that distinction we ignore it as well. */
-    rc = GetEnvironmentVariableA(variable, buf, bufsize);
+    rc = GetEnvironmentVariableW(variable, buf, bufsize);
     if(!rc || rc == bufsize || rc > max) {
       curlx_free(buf);
       return NULL;
@@ -62,6 +59,27 @@ char *curl_getenv(const char *variable)
 
     /* else rc is bytes needed, try again */
   }
+}
+#endif
+
+char *curl_getenv(const char *variable)
+{
+#if defined(CURL_WINDOWS_UWP) || \
+  defined(__ORBIS__) || defined(__PROSPERO__) /* PlayStation 4 and 5 */
+  (void)variable;
+  return NULL;
+#elif defined(_WIN32)
+  char *value = NULL;
+  wchar_t *variable_w = curlx_convert_UTF8_to_wchar(variable);
+  if(variable_w) {
+    wchar_t *value_w = GetEnv(variable_w);
+    if(value_w) {
+      value = curlx_convert_wchar_to_UTF8(value_w);
+      curlx_free(value_w);
+    }
+    curlx_free(variable_w);
+  }
+  return value;
 #else
   char *env = getenv(variable);
   return (env && env[0]) ? curlx_strdup(env) : NULL;
