 src/tool_getparam.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/tool_getparam.c b/src/tool_getparam.c
index 40ac4e3..a5ed701 100644
--- a/src/tool_getparam.c
+++ b/src/tool_getparam.c
@@ -23,6 +23,10 @@
  ***************************************************************************/
 #include "tool_setup.h"
 
+#ifndef HAVE_WRITABLE_ARGV
+#define HAVE_WRITABLE_ARGV 1
+#endif
+
 #include "strcase.h"
 
 #include "curlx.h"
@@ -558,8 +562,13 @@ static void cleanarg(argv_item_t str)
    * argument out so that the username:password is not displayed in the
    * system process list */
   if(str) {
+#ifdef _UNICODE
+    size_t len = wcslen(str);
+    wmemset(str, L' ', len);
+#else
     size_t len = strlen(str);
     memset(str, ' ', len);
+#endif
   }
 }
 #else
