 configure        | 2 +-
 lib/curl_setup.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 42bde11..a04bbbf 100644
--- a/configure
+++ b/configure
@@ -18616,7 +18616,7 @@ printf %s "checking if building for Windows Vista or newer... " >&6; }
 int main(void)
 {
 
-      #if (_WIN32_WINNT < 0x600)
+      #if (_WIN32_WINNT < 0x501)
       #error
       #endif
 
diff --git a/lib/curl_setup.h b/lib/curl_setup.h
index 4f8d65a..8327c68 100644
--- a/lib/curl_setup.h
+++ b/lib/curl_setup.h
@@ -160,7 +160,7 @@
 #endif /* HAVE_CONFIG_H */
 
 #ifdef _WIN32
-#  if defined(_WIN32_WINNT) && (_WIN32_WINNT < 0x0600)
+#  if defined(_WIN32_WINNT) && (_WIN32_WINNT < 0x0501)
 #    error The minimum build target is Windows Vista (0x0600)
 #  endif
 
