 lib/curl_setup.h | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/lib/curl_setup.h b/lib/curl_setup.h
index 9e293c2..be1d2fa 100644
--- a/lib/curl_setup.h
+++ b/lib/curl_setup.h
@@ -681,14 +681,17 @@
 #endif /* _WIN32 */
 
 /* We want to use mutex when available. */
-#if defined(HAVE_THREADS_POSIX) || defined(_WIN32)
-#define USE_MUTEX
-#endif
+//#if defined(HAVE_THREADS_POSIX) || defined(_WIN32)
+//#define USE_MUTEX
+//#endif
+#undef USE_MUTEX
 
 /* threaded resolver is the only feature requiring threads. */
-#ifdef USE_RESOLV_THREADED
-#define USE_THREADS
-#endif
+//#ifdef USE_RESOLV_THREADED
+//#define USE_THREADS
+//#endif
+#undef USE_THREADS
+#undef USE_RESOLV_THREADED
 
 /* ---------------------------------------------------------------- */
 /*             resolver specialty compile-time defines              */
