 src/tool_operate.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/tool_operate.c b/src/tool_operate.c
index 32858f114..463630818 100644
--- a/src/tool_operate.c
+++ b/src/tool_operate.c
@@ -2111,11 +2111,15 @@ static CURLcode cacertpaths(struct OperationConfig *config)
 #ifdef _WIN32
   if(!env) {
 #if defined(CURL_CA_SEARCH_SAFE)
-    char *cacert = NULL;
-    FILE *cafile = tool_execpath("curl-ca-bundle.crt", &cacert);
+    TCHAR *cacert = NULL;
+    FILE *cafile = tool_execpath(TEXT("curl-ca-bundle.crt"), &cacert);
     if(cafile) {
       fclose(cafile);
+#ifdef UNICODE
+      config->cacert = curlx_convert_wchar_to_UTF8(cacert);
+#else
       config->cacert = strdup(cacert);
+#endif
     }
 #elif !defined(CURL_WINDOWS_UWP) && !defined(UNDER_CE) && \
   !defined(CURL_DISABLE_CA_SEARCH)
