 src/tool_getparam.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/tool_getparam.c b/src/tool_getparam.c
index 6be57dbd5..2d05450b3 100644
--- a/src/tool_getparam.c
+++ b/src/tool_getparam.c
@@ -43,6 +43,10 @@
 #define ALLOW_BLANK TRUE
 #define DENY_BLANK FALSE
 
+#ifndef HAVE_WRITABLE_ARGV
+#define HAVE_WRITABLE_ARGV 1
+#endif
+
 static ParameterError getstr(char **str, const char *val, bool allowblank)
 {
   if(*str) {
@@ -572,14 +576,19 @@ static ParameterError GetSizeParameter(const char *arg,
 }
 
 #ifdef HAVE_WRITABLE_ARGV
-static void cleanarg(char *str)
+static void cleanarg(TCHAR *str)
 {
   /* now that getstr has copied the contents of nextarg, wipe the next
    * argument out so that the username:password is not displayed in the
    * system process list */
   if(str) {
+#ifdef _UNICODE
+    size_t len = wcslen(str);
+    wmemset(str, L' ', len);
+#else
     size_t len = strlen(str);
     memset(str, '*', len);
+#endif
   }
 }
 #else
