diff --git "a/D:\\_Sandbox\\curl\\libgsasl-src\\src\\error.c.orig" "b/D:\\_Sandbox\\curl\\libgsasl-src\\src\\error.c"
index e61b945..819ba8e 100644
--- "a/src/error.c"
+++ "b/src/error.c"
@@ -23,10 +23,11 @@
 #include "internal.h"
 
 /* I18n of error codes. */
-#include "gettext.h"
-#define _(String) dgettext (PACKAGE, String)
+/* #include "gettext.h" */
+#define _(String) (String)
 #define gettext_noop(String) String
 #define N_(String) gettext_noop (String)
+#define bindtextdomain(P,L) ((void)0)
 
 #define ERR(name, desc) { name, #name, desc }
 
