 src/ssl.c        | 2 +-
 wolfssl/wolfio.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ssl.c b/src/ssl.c
index 0b74065..2800af3 100644
--- a/src/ssl.c
+++ b/src/ssl.c
@@ -22108,7 +22108,7 @@ unsigned long wolfSSL_ERR_peek_error_line_data(const char **file, int *line,
 #if defined(OPENSSL_EXTRA) || defined(OPENSSL_ALL) || \
     defined(WOLFSSL_NGINX) || defined(WOLFSSL_HAPROXY)
 
-#if !defined(WOLFSSL_USER_IO)
+#if 0
 /* converts an IPv6 or IPv4 address into an octet string for use with rfc3280
  * example input would be "127.0.0.1" and the returned value would be 7F000001
  */
diff --git a/wolfssl/wolfio.h b/wolfssl/wolfio.h
index de45a18..2c88848 100644
--- a/wolfssl/wolfio.h
+++ b/wolfssl/wolfio.h
@@ -908,7 +908,7 @@ WOLFSSL_API void wolfSSL_SetIOWriteFlags(WOLFSSL* ssl, int flags);
         #define XINET_NTOP(a,b,c,d) inet_ntop((a),(b),(c),(d))
     #endif
 #endif
-#ifndef XINET_PTON
+#if 0
     #if defined(__WATCOMC__)
         #if defined(__OS2__) || defined(__NT__) && \
                 (NTDDI_VERSION >= NTDDI_VISTA)
